src/dns/aardvark.rs

```diff
@@ -344,8 +344,18 @@ impl Aardvark {
 
     pub fn commit_netavark_entries(&self, entries: Vec<AardvarkEntry>) -> NetavarkResult<()> {
         if !entries.is_empty() {
-            self.commit_entries(entries)?;
-            self.notify(true, false)?;
+            self.commit_entries(&entries)?;
```

Comment:
Nit and non-blocking:

Not sure can this be simplified using `map_err` ? Although current code has better readability.

```diff
diff --git a/src/dns/aardvark.rs b/src/dns/aardvark.rs
index c1a967e..ad97a55 100644
--- a/src/dns/aardvark.rs
+++ b/src/dns/aardvark.rs
@@ -345,17 +345,12 @@ impl Aardvark {
     pub fn commit_netavark_entries(&self, entries: Vec<AardvarkEntry>) -> NetavarkResult<()> {
         if !entries.is_empty() {
             self.commit_entries(&entries)?;
-            match self.notify(true, false) {
-                Ok(_) => (),
-                Err(e) => {
-                    if let Err(err) = self.delete_from_netavark_entries(&entries) {
-                        log::warn!(
-                            "Failed to delete aardvark-dns entries after failed start: {err}"
-                        );
-                    };
-                    return Err(e);
+            self.notify(true, false).map_err(|e| {
+                if let Err(err) = self.delete_from_netavark_entries(&entries) {
+                    log::warn!("Failed to delete aardvark-dns entries after failed start: {err}");
                 }
-            };
+                e
+            })?;
         }
         Ok(())
     }
```

Reply:
I think map_err is best used only when you modify the error and not add extra logic so I think the match case makes it more clear what is going on.

Reply:
Fair enough.